### ISSUE TITLE
UnboundLocalError: local variable 'y_align' referenced before assignment

### ISSUE THREAD

**Issue by MengPhantom** [Kozea/CairoSVG#158]
I use cairosvg parse a svgz file ,and got this error.how can I fix it 
```
---> 21     bytestr = svg2png(bytestring=text,parent_width=width, parent_height=height)
     22     io = StringIO(bytestr)
     23     im = Image.open(io)

/Users/a024/env/lib/python2.7/site-packages/cairosvg/__init__.pyc in <lambda>(*args, **kwargs)
     41         # Two lambdas needed for the closure
     42         lambda surface_type: lambda *args, **kwargs:  # pylint: disable=W0108
---> 43             surface_type.convert(*args, **kwargs))(_surface_type)
     44     _name = 'svg2%s' % _output_format.lower()
     45     _function.__name__ = _name

/Users/a024/env/lib/python2.7/site-packages/cairosvg/surface/__init__.pyc in convert(cls, bytestring, **kwargs)
    106         else:
    107             output = write_to
--> 108         cls(tree, output, dpi, None, parent_width, parent_height).finish()
    109         if write_to is None:
    110             return output.getvalue()

/Users/a024/env/lib/python2.7/site-packages/cairosvg/surface/__init__.pyc in __init__(self, tree, output, dpi, parent_surface, parent_width, parent_height)
    159         self.set_context_size(width, height, viewbox)
    160         self.context.move_to(0, 0)
--> 161         self.draw_root(tree)
    162
    163     @property

/Users/a024/env/lib/python2.7/site-packages/cairosvg/surface/__init__.pyc in draw_root(self, node)
    212     def draw_root(self, node):
    213         """Draw the root ``node``."""
--> 214         self.draw(node)
    215
    216     def draw(self, node):

/Users/a024/env/lib/python2.7/site-packages/cairosvg/surface/__init__.pyc in draw(self, node)
    383                 "mask", "clipPath", "filter"):
    384             for child in node.children:
--> 385                 self.draw(child)
    386
    387         if filter_ or mask or (opacity < 1 and node.children):

/Users/a024/env/lib/python2.7/site-packages/cairosvg/surface/__init__.pyc in draw(self, node)
    383                 "mask", "clipPath", "filter"):
    384             for child in node.children:
--> 385                 self.draw(child)
    386
    387         if filter_ or mask or (opacity < 1 and node.children):

/Users/a024/env/lib/python2.7/site-packages/cairosvg/surface/__init__.pyc in draw(self, node)
    326         if node.tag in TAGS:
    327             try:
--> 328                 TAGS[node.tag](self, node)
    329             except PointError:
    330                 # Error in point parsing, do nothing

/Users/a024/env/lib/python2.7/site-packages/cairosvg/surface/text.pyc in text(surface, node)
    203                 cursor_position = x + letter_spacing + extents, y
    204                 surface.context.rel_move_to(*surface.cursor_d_position)
--> 205                 surface.context.rel_move_to(-x_align, y_align)
    206                 surface.context.rotate(last_r if r is None else r)
    207

UnboundLocalError: local variable 'y_align' referenced before assignment
```

**Comment by liZe** [comment]
1. Thanks for this bug report.
2. Could you try a recent version of CairoSVG? Python 2's support has been dropped since 2.0.0.
2. If you have the same problem with a recent version, could you please provide your SVG file?

**Comment by MengPhantom** [comment]
mac py2cairo

**Comment by GeibTobias** [comment]
Hello, I have a similar issue when i try to call `svg2png()`.

I am on Windows, using Python 3.6.

When i call `cairosvg.svg2pdf(file_obj=open('page.svg', "rb"), write_to='page.pdf')`, i get the following traceback:

```
Traceback (most recent call last):
  File "C:/Users/.../main.py", line 28, in <module>
    write_to=r'C:\Users\...\page.png')
  File "C:\Program Files\Python36\lib\site-packages\cairosvg\__init__.py", line 45, in <lambda>
    surface_type.convert(*args, **kwargs))(_surface_type)
  File "C:\Program Files\Python36\lib\site-packages\cairosvg\surface.py", line 138, in convert
    tree, output, dpi, None, parent_width, parent_height, scale)
  File "C:\Program Files\Python36\lib\site-packages\cairosvg\surface.py", line 195, in __init__
    self.draw(tree)
  File "C:\Program Files\Python36\lib\site-packages\cairosvg\surface.py", line 413, in draw
    self.draw(child)
  File "C:\Program Files\Python36\lib\site-packages\cairosvg\surface.py", line 413, in draw
    self.draw(child)
  File "C:\Program Files\Python36\lib\site-packages\cairosvg\surface.py", line 413, in draw
    self.draw(child)
  [Previous line repeated 1 more times]
  File "C:\Program Files\Python36\lib\site-packages\cairosvg\surface.py", line 353, in draw
    TAGS[node.tag](self, node)
  File "C:\Program Files\Python36\lib\site-packages\cairosvg\text.py", line 203, in text
    surface.context.rel_move_to(-x_align, y_align)
UnboundLocalError: local variable 'y_align' referenced before assignment
```
When i try to call `cairosvg.svg2pdf(file_obj=open('page.svg', "rb"), write_to='page.pdf')`, I get the following traceback:

```
  File "C:/Users/.../main.py", line 28, in <module>
    write_to=r'C:\Users\...\page.pdf')
  File "C:\Program Files\Python36\lib\site-packages\cairosvg\__init__.py", line 45, in <lambda>
    surface_type.convert(*args, **kwargs))(_surface_type)
  File "C:\Program Files\Python36\lib\site-packages\cairosvg\surface.py", line 138, in convert
    tree, output, dpi, None, parent_width, parent_height, scale)
  File "C:\Program Files\Python36\lib\site-packages\cairosvg\surface.py", line 195, in __init__
    self.draw(tree)
  File "C:\Program Files\Python36\lib\site-packages\cairosvg\surface.py", line 413, in draw
    self.draw(child)
  File "C:\Program Files\Python36\lib\site-packages\cairosvg\surface.py", line 413, in draw
    self.draw(child)
  File "C:\Program Files\Python36\lib\site-packages\cairosvg\surface.py", line 413, in draw
    self.draw(child)
  File "C:\Program Files\Python36\lib\site-packages\cairosvg\surface.py", line 353, in draw
    TAGS[node.tag](self, node)
  File "C:\Program Files\Python36\lib\site-packages\cairosvg\text.py", line 81, in text
    surface.context.font_extents())
  File "C:\Program Files\Python36\lib\site-packages\cairocffi\context.py", line 1929, in font_extents
    self._check_status()
  File "C:\Program Files\Python36\lib\site-packages\cairocffi\context.py", line 108, in _check_status
    _check_status(cairo.cairo_status(self._pointer))
  File "C:\Program Files\Python36\lib\site-packages\cairocffi\__init__.py", line 79, in _check_status
    raise exception(message, status)
cairocffi.CairoError: cairo returned 41: b'error occurred in the Windows Graphics Device Interface'
```

A pdf file is actually created, but it only renders a portion of the SVG.
I have attached a zip-file with both page.svg and page.pdf.
[svgbug.zip](https://github.com/Kozea/CairoSVG/files/1996006/svgbug.zip)

Strangely, when i try another SVG, it works as it's supposed to.

**Comment by liZe** [comment]
It's fixed by #184.
